packages/teams-js/test/public/webStorage.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { cli, version } from 'webpack-dev-middleware/node_modules/webpack';
+import { app } from '../../src/public/app';
+import { FrameContexts, HostClientType } from '../../src/public/constants';
+import { _minRuntimeConfigToUninitialize } from '../../src/public/runtime';
+import { webStorage } from '../../src/public/webStorage';
+import { FramelessPostMocks } from '../framelessPostMocks';
+import { Utils } from '../utils';
+import { generateBackCompatRuntimeConfig } from '../../src/public/runtime';
+import { compareSDKVersions } from '../../src/internal/utils';
+
+describe('webStorage', () => {
+  const framelessPlatformMock = new FramelessPostMocks();
+  const framedPlatformMock = new Utils();
+  const testVersions = ['1.8.0', '2.0.4', '2.0.5', '2.0.6'];
+  const minMobileVersionForWebStorage = '2.0.5';
+
+  beforeEach(() => {
+    framelessPlatformMock.messages = [];
+
+    // Set a mock window for testing
+    app._initialize(framelessPlatformMock.mockWindow);
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (app._uninitialize) {
+      framedPlatformMock.setRuntimeConfig(_minRuntimeConfigToUninitialize);
+      framedPlatformMock.setRuntimeConfig(_minRuntimeConfigToUninitialize);
+      app._uninitialize();
+    }
+    jest.clearAllMocks();
+  });
+
+  describe('isWebStorageClearedOnUserLogOut', () => {
+    it('should not allow calls before initialization', () => {
+      expect(webStorage.isWebStorageClearedOnUserLogOut).toThrowError('The library has not yet been initialized');
+    });
+    Object.values(HostClientType).forEach(clientType => {
+      Object.values(FrameContexts).forEach(context => {
+        Object.values(testVersions).forEach(version => {
+        it(`should allow Framed - isWebStorageClearedOnUserLogOut calls for version ${version} and context ${context} and hostClientType ${clientType}`, async () => {
```

Comment:
The way these tests are structured does not make it easy to read, nor accurately conveys what each test is doing.

The goal: each test case should be printed to the console with the name of what the method is testing along with the interesting parameters and the expected output.
It's also helpful if all the framed scenarios are grouped together, followed by all the positive test cases and within the positive test cases, the varying parameter input values. Then all the framed scenarios with negative test cases and all the varying  parameter input values. Then all the frameless tests that are positive and all the frameless tests that are negative.

The way it's currently structured, it prints that it should allow `isWebStorageClearedOnUserLogOut` calls for every test, even the ones for which it should not.

It's also harder to parse the output when you're reading it.

Suggestion:
Break up this code into separate describe sections, e.g.,
describe("Framed - isWebStorageClearedOnUserLogOut")

and then within that, group the positive test cases first followed by the negative test cases.

I sent you examples offline.

Also, you can run the tests and see detailed output in packages/teams-js by using this command `yarn test tests/public/webStorage.spec.ts`